### PR TITLE
Ee 9536

### PR DESCRIPTION
**Pull request by Tim-Hammonds**
Bit of a tidy up.
Added first of the executable specs to replace the Integration tests

### CODE REVIEW

File: src/test/resources/specs/name-matching/Last_names_combination.feature [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+      | First name  | Last name     |
+      | Arthur      | Coates        |
+      | Brian       | Coates        |
+      | Coates      | Arthur        |
+      | Coates      | Brian         |
+      | Brian       | Arthur        |
+
+
+  @name_matching
+  Scenario: Multi-part first name is tried without splitting
```
Comment (owainrobertsbjss) [inline src/test/resources/specs/name-matching/Last_names_combination.feature]: In the candidate generators multi-part means more than one word separated by spaces. Although the result will be the same, I think this test would be better if "Arthur-Brian" was "Arthur Brian", unless there's a reason for hyphenating?

If that's how it was in the unit test then maybe the title should be changed, and maybe we should test both cases.

Comment (Tim-Hammonds) [reply src/test/resources/specs/name-matching/Last_names_combination.feature]: Modifying the terminology. Split first name and last name tests into separate feature files.

Comment (Tim-Hammonds) [reply src/test/resources/specs/name-matching/Last_names_combination.feature]: I've clarified the test name and what it tests

File: src/test/java/bdd/steps/NameMatchingSteps.java [hunk 2]
```diff
@@ -231,79 +232,148 @@ public void theFootprintWillTryTheFollowingCombinationOfNamesInOrder(DataTable d
...
+
+        return names
+                   .stream()
+                   .map(produceJsonForName)
+                   .map(this::produceExpectedNameMatcher)
+                   .map(expectedNameMatcher -> loggedRequests
+                                                         .stream()
+                                                         .map(LoggedRequest::getBodyAsString)
+                                                         .anyMatch(requestBody -> expectedNameMatcher.match(requestBody).isExactMatch()))
+                   .filter(this::aRequestWasMadeForThisName)
```
Comment (owainrobertsbjss) [inline src/test/java/bdd/steps/NameMatchingSteps.java]: As we have discussed `.filter(m -> m == true)` is probably the clearest way to express this.

**Pull request merged**